packages/frontend/editor-ui/src/composables/useWorkflowHelpers.ts

```diff
@@ -71,20 +75,61 @@ export type ResolveParameterOptions = {
 
 export function resolveParameter<T = IDataObject>(
 	parameter: NodeParameterValue | INodeParameters | NodeParameterValue[] | INodeParameters[],
+	opts: ResolveParameterOptions | ExpressionLocalResolveContext = {},
+): T | null {
+	if ('localResolve' in opts && opts.localResolve) {
+		return resolveParameterImpl(
+			parameter,
+			opts.workflow,
+			opts.envVars,
+			opts.workflow.getNode(opts.nodeName),
+			opts.execution,
+			true,
+			opts.workflow.pinData,
+			{
+				inputNodeName: opts.inputNode?.name,
+				inputRunIndex: opts.inputNode?.runIndex,
+				inputBranchIndex: opts.inputNode?.branchIndex,
+				additionalKeys: opts.additionalKeys,
+			},
+		);
+	}
+
+	const workflowsStore = useWorkflowsStore();
+
+	return resolveParameterImpl(
+		parameter,
+		workflowsStore.getCurrentWorkflow(),
+		useEnvironmentsStore().variablesAsObject,
+		useNDVStore().activeNode,
+		workflowsStore.workflowExecutionData,
+		workflowsStore.shouldReplaceInputDataWithPinData,
+		workflowsStore.pinnedWorkflowData,
+		opts,
+	);
+}
+
+// TODO: move to separate file
```

Comment:
I want to move this to a separate utility file to emphasis that this doesn't depend on global state, but will do in a separate PR